src/ppom-controller.ts

```diff
@@ -332,25 +342,42 @@ export class PPOMController extends BaseControllerV2<
    * This function receives a callback that will be called with the PPOM.
    *
    * @param callback - Callback to be invoked with PPOM.
+   * @param networkClientId - Optional network client ID to use for the PPOM requests.
    */
   async usePPOM<T>(
-    callback: (ppom: any) => Promise<T>,
+    callback: (ppom: unknown) => Promise<T>,
+    networkClientId?: NetworkClientId,
   ): Promise<T & { providerRequestsCount: Record<string, number> }> {
     if (!this.#securityAlertsEnabled) {
       throw Error('User has securityAlertsEnabled set to false');
     }
-    if (!this.#networkIsSupported(this.#chainId)) {
-      throw Error('Blockaid validation is available only on ethereum mainnet');
+
+    let chainId = this.#chainId;
+    let provider = this.#provider;
+    if (networkClientId) {
+      const networkClient = this.messagingSystem.call(
+        'NetworkController:getNetworkClientById',
+        networkClientId,
+      );
+      chainId = networkClient.configuration.chainId;
+      provider = networkClient.provider;
     }
+    this.#updateChainStatus(chainId);
 
-    await this.#getNewFilesForCurrentChainIfNeeded();
+    if (!this.#networkIsSupported(chainId)) {
+      throw Error('Blockaid validation is available only on ethereum mainnet');
+    }
 
-    this.#providerRequests = 0;
-    this.#providerRequestsCount = {};
+    await Promise.all([
+      this.#getNewFilesForChainIfNeeded(chainId),
+      this.#initialisePPOM(),
+    ]);
```

Comment:
Just noting that previously these were previously awaited sequentially? I'm assuming there is no issue with running them in parallel? No race conditions here? From a cursory glance there doesn't appear to be but just checking if someone with a deeper understanding of this controller might know?

Reply:
correct. these tasks can be completed concurrently